src/PactNet.Native/NativePactMessageBuilder.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Generic;
+
+using Newtonsoft.Json;
+
+using PactNet.Native.Exceptions;
+using PactNet.Native.Models;
+
+namespace PactNet.Native
+{
+    /// <summary>
+    /// Mock request builder
+    /// </summary>
+    public class NativePactMessageBuilder : IPactMessageBuilderV3
+    {
+        private readonly IMessageMockServer server;
+        private readonly MessagePactHandle pact;
+        private readonly PactConfig config;
+        private readonly MessageHandle message;
+        private readonly JsonSerializerSettings defaultSettings;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="NativePactMessageBuilder"/> class.
+        /// </summary>
+        /// <param name="server">Mock server</param>
+        /// <param name="pact">the pact handle</param>
+        /// <param name="config">the pact configuration</param>
+        /// <param name="defaultSettings">Default JSON serializer settings</param>
+        internal NativePactMessageBuilder(IMessageMockServer server, MessagePactHandle pact, PactConfig config, JsonSerializerSettings defaultSettings = null)
+        {
+            this.pact = pact;
+            this.config = config ?? throw new ArgumentNullException(nameof(config));
+            this.server = server ?? throw new ArgumentNullException(nameof(server));
+            message = server.NewMessage(pact, "default message");
```

Comment:
This call shouldn't be done in the constructor because it can fail.

Reply:
With the builder state pattern, it is now done by the ExpectedToReceive before it creates the message via another builder.